internal/gcs/container.go

```diff
@@ -224,6 +224,10 @@ func (c *Container) Terminate(ctx context.Context) (err error) {
 	return c.shutdown(ctx, rpcShutdownForced)
 }
 
+func (c *Container) WaitChannel() <-chan struct{} {
+	return c.notifyCh
```

Comment:
I think we should maintain parity with `Wait` here regarding `closeCh`, to be on the safe side. I would suggest the following:

Add to `Container`:
```golang
waitCh chan struct{}
waitErr error
```

Add to `CreateContainer`/`CloneContainer`:
```golang
go funct() {
    select {
    case <-c.notifyCh:
    case <-c.closeCh:
        c.waitErr = errors.New("container closed")
    }
    close(c.waitCh)
}()
```

Add here:
```golang
return c.waitCh
```

Reply:
This would mean each container has a single background goroutine responsible for waiting on `notifyCh`/`closeCh`, setting `waitErr`, and closing `waitCh`.

And then we can simply return `waitCh` as the unified waiting channel from here.

Reply:
Actually, we already have `waitBackground` that doesn't do anything except log, so we can use it as our background goroutine for this wait too.